packages/babel-plugin/src/utils/build-compiled-component.ts

```diff
@@ -57,6 +58,49 @@ const getExpression = (
   return value;
 };
 
+/**
+ * Check if 'node' is a 'inBuiltComponent'.
+ * We assume it's a 'inBuiltComponent' if it starts with a lowercase letter (i.e. div).
+ *
+ * @param node Originating node
+ */
+const isInBuiltComponent = (node: t.JSXElement): boolean => {
+  let tagName;
+
+  // Get the tag name (i.e. td) from node <td />
+  if (
+    t.isJSXIdentifier(node.openingElement.name) ||
+    t.isJSXNamespacedName(node.openingElement.name)
+  ) {
+    tagName = node.openingElement.name.name;
+  } else {
+    tagName = node.openingElement.name.property.name;
+  }
+
+  if (typeof tagName !== 'string') return false;
+
+  // Check if the first letter is lower case
+  return tagName.charCodeAt(0) >= 97 && tagName.charCodeAt(0) <= 122;
+};
+
+/**
+ * Check if the `ax` runtime utility is required.
+ * We assume it's required if
+ * - there is a class name prop statically defined
+ * - there is conditional css
+ * - 'node' is a 'UserDefinedComponent'
```

Comment:
props spreading is also another way for the component to have a defined classname

Reply:
Good point! 👍

- [ ] consider props spreading 

I notice props spreading can cause issue in existing/master codebase too. Raised an issue: https://github.com/atlassian-labs/compiled/issues/1361